f5lbaasdriver/test/tempest/tests/api/test_l7policy_rules.py

```diff
@@ -65,51 +70,119 @@ def resource_setup(cls):
         cls.pool_id = cls.pool['id']
 
         # Create basic args for policy creation
-        create_l7policy_rule_kwargs = {'listener_id': cls.listener_id,
-                                       'action': 'REJECT',
-                                       'admin_state_up': 'true'}
-        new_policy = cls._create_l7policy(
-            **create_l7policy_rule_kwargs)
-        cls.policy_id = new_policy['id']
+        cls.create_l7policy_kwargs = {'listener_id': cls.listener_id,
+                                      'admin_state_up': "true"}
 
         # Get a client to emulate the agent's behavior.
         cls.client = cls.plugin_rpc.get_client()
         cls.context = cls.plugin_rpc.get_context()
 
-    @classmethod
-    def resource_cleanup(cls):
-        super(L7PolicyRulesTestJSON, cls).resource_cleanup()
-
-    def check_rules(self):
-        # Check service object has rules we expect
-        res = self.client.call(self.context, "get_service_by_loadbalancer_id",
-                               loadbalancer_id=self.load_balancer_id)
-        assert 'l7policy_rules' in res.keys()
-        print(res['l7policy_rules'])
-        assert len(res['l7policy_rules']) == 1
-        rule = res['l7policy_rules'][0]
-        assert rule['policy_id'] == self.policy_id
-
-    def set_agent_calls(self, rule_id):
-        # These actions should be performed by the agent.
-        self.client.call(self.context, "update_l7rule_status",
-                         l7rule_id=rule_id,
-                         l7policy_id=self.policy_id,
-                         provisioning_status=plugin_const.ACTIVE)
-        self.client.call(self.context, "update_l7policy_status",
-                         l7policy_id=self.policy_id,
-                         provisioning_status=plugin_const.ACTIVE)
-        self.client.call(self.context, "update_loadbalancer_status",
-                         loadbalancer_id=self.load_balancer_id,
-                         status=plugin_const.ACTIVE,
-                         operating_status=lb_const.ONLINE)
 
+
+
+
+class TestL7PolicyTestJSONReject(L7PolicyTestJSONBasic):
     @test.attr(type='smoke')
-    def test_create_l7_starts_with_rule(self):
+    def test_create_l7_reject_policy(self):
         """Test the creationg of a L7 reject policy."""
-        l7rule = self._create_l7rule(
-            self.policy_id,
-            type='PATH', compare_type='STARTS_WITH', value='/api')
-        self.set_agent_calls(l7rule['id'])
-        self.addCleanup(self._delete_l7rule, self.policy_id, l7rule['id'])
-        self.check_rules()
+        create_l7policy_kwargs = self.create_l7policy_kwargs
+        create_l7policy_kwargs['action'] = "REJECT"
+
+        new_policy = self._create_l7policy(
+            **create_l7policy_kwargs)
+        self._wait_for_load_balancer_status(self.load_balancer_id)
+        self.addCleanup(self._delete_l7policy, new_policy['id'])
+        assert(new_policy['action'] == "REJECT")
+
```

Comment:
Before the end of the test, I would assert that no policy exists on the BIG-IP for this tenant.